src/screens/CreatePost/index.js

```diff
@@ -140,13 +140,18 @@ const CreatePost = () => {
   const animatedReminder = React.useRef(new Animated.Value(0)).current;
   const debounced = React.useCallback(
     debounce((changedText) => {
+      if (mediaStorage.length > 0) {
+        setIsLinkPreviewShown(false);
+        return;
+      }
+
       if (isContainUrl(changedText)) {
         getPreviewUrl(getUrl(changedText));
       } else {
         setIsLinkPreviewShown(false);
       }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here seems fine, but it would be better to use a more descriptive function name than `isContainUrl`. Consider renaming it to `containsUrl` or `doesContainUrl` for better readability.

154:
The dependency array for `useCallback` now includes `mediaStorage`. This is a good change as it ensures that the callback is updated whenever `mediaStorage` changes. However, if `mediaStorage` is a complex object or array, this could lead to unnecessary re-renders as `useCallback` will create a new function whenever the reference to `mediaStorage` changes, even if the actual data hasn't changed. If this becomes a performance issue, consider using a library like lodash's `isEqual` function in a custom hook to compare the previous and current values of `mediaStorage` before deciding whether to update the callback.


<!-- This is an auto-generated comment by OSS CodeRabbit -->